hashable.cabal

```diff
@@ -64,6 +64,13 @@ flag random-initial-seed
   manual:      True
   default:     False
 
+flag os-string
+  description:
+    Use the new os-string package and filepath pacakage version >= 1.5.0.0
+
+  manual:      False
+  default:     False
```

Comment:
`default:     False` here interacts badly with `--allow-newer`. Trying to do any builds that use that will result in our flag being false but using a version of filepath that needs it.

Reply:
There's no way around it. It has to be an automatic flag.

Reply:
I know it has to be automatic, but can't it be default-true automatic?

Reply:
I find it hard to reason about this. Also note that all the other packages have the default as `False`.

This strategy is also suggested by Cabal-syntax, which we derived this from: https://hackage.haskell.org/package/Cabal-syntax-3.6.0.0

Maybe @Bodigrim knows more.

Reply:
@josephcsible if you are already passing `--allow-newer` you can pass `-c 'hashable +os-string'` as well.

Reply:
@Bodigrim I know I can do that, but I don't want to have to do that for every single `cabal install --allow-newer` that I ever run in the future that pulls in hashable.

Reply:
Another idea: if we really want to keep the flag `default: False`, then could we do [what QuickCheck does](https://github.com/nick8325/quickcheck/blob/e703e6882fb41ce0b470be6d965dd665cc6a1fe2/QuickCheck.cabal#L75-L90), calling the flag something like `old-filepath`, and having `True` be the legacy path and `False` be the modern path?

Reply:
I don't know any particular reason to stick to `default: False`, but I also find optimizing for `--allow-newer` to be a weak justification. 

Ultimately, I think that

> dependency on `os-string` should be unconditional (and code to be written).

is a better solution, in which case the default value of the flag would be immaterial for `--allow-newer`.

Reply:
The solution that was just merged in #293 doesn't cause this problem, so I'm happy with it.